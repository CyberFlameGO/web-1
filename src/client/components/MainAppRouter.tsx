import * as React from 'react';
import { BrowserRouter as Router, Route, Switch, RouteComponentProps } from 'react-router-dom';

import { MenuHeader } from './MenuHeader';
import { Dashboard } from './Dashboard';
import { ProjectConfig } from './ProjectConfig';
import { SlackOAuthResult } from './SlackOAuthResult';

function NotFoundHandler(props: RouteComponentProps<any>) {
  React.useEffect(() => {
    props.history.replace('/');
  }, [props.location.pathname]);

  return <h1>Route Not Found</h1>;
}

export function MainAppRouter() {
  return (
    <Router>
      <MenuHeader />
      <Switch>
        <Route path="/" exact component={Dashboard} />
        <Route path="/project/:projectId" exact component={ProjectConfig} />
        <Route path="/oauth_result/slack" exact component={SlackOAuthResult} />
        <Route component={NotFoundHandler} />
      </Switch>
    </Router>
  );
}
