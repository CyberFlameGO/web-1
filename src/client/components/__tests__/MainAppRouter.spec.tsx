import * as React from 'react';
import { shallow } from 'enzyme';
import { MainAppRouter } from '../MainAppRouter';

describe('<MainAppRouter />', () => {
  it('Should render a router', () => {
    const wrapper = shallow(<MainAppRouter />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('BrowserRouter')).toHaveLength(1);
  });
});
