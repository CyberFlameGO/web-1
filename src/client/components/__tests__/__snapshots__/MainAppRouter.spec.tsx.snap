// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainAppRouter /> Should render a router 1`] = `
<BrowserRouter>
  <withRouter(MenuHeaderInner) />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/project/:projectId"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/oauth_result/slack"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</BrowserRouter>
`;
